root/account/Donation.js

```diff
@@ -9,55 +9,76 @@
 
 import * as React from 'react';
 
+import UserAccountLayout, {
+  sanitizedAccountLayoutUser,
+} from '../components/UserAccountLayout';
 import {CONTACT_URL, DONATE_URL} from '../constants';
-import StatusPage from '../components/StatusPage';
+import Warning from '../static/scripts/common/components/Warning';
 
 type Props = {
+  +checkFailed: boolean,
   +days: number,
   +nag: boolean,
+  +user: UnsanitizedEditorT,
 };
 
-const Donation = ({days, nag}: Props): React.Element<typeof StatusPage> => (
-  <StatusPage title={l('Donation Check')}>
-    {nag
-      ? (
-        <>
-          <p>
-            {l(`We have not received a donation from you recently. If you have
-                just made a PayPal donation, then we have not received a
-                notification from PayPal yet. Please wait a few minutes and
-                reload this page to check again.`)}
-          </p>
-          <p>
-            {exp.l(
-              `If you would like to make a donation,
-               {donate|you can do that here}. If you have donated, but
-               you are still being nagged, please {contact|contact us}.`,
-              {contact: CONTACT_URL, donate: DONATE_URL},
-            )}
-          </p>
-        </>
-      ) : (
-        <>
-          <p>
-            {l('Thank you for contributing to MusicBrainz.')}
-          </p>
-          {days > 0
-            ? (
-              <p>
-                {texp.l(
-                  'You will not be nagged for another {days} days.',
-                  {days: days},
-                )}
-              </p>
-            ) : (
-              <p>
-                {l('You will never be nagged again!')}
-              </p>
-            )}
-        </>
-      )}
-  </StatusPage>
+const Donation = ({
+  checkFailed,
+  days,
+  nag,
+  user,
+}: Props): React.Element<typeof UserAccountLayout> => (
+  <UserAccountLayout
+    entity={sanitizedAccountLayoutUser(user)}
+    page="donation"
+    title={l('Donation Check')}
+  >
+    <h2>{l('Donation Check')}</h2>
+    {checkFailed ? (
+      <Warning
```

Comment:
This is more like an error than a warning, but not sure we have anything better at the moment.

Reply:
It just felt better than a 500 with a message :)